LeaderboardBackend/Controllers/RunsController.cs

```diff
@@ -157,4 +157,51 @@ public async Task<ActionResult<CategoryViewModel>> GetCategoryForRun(Guid id)
 
         return Ok(CategoryViewModel.MapFrom(category));
     }
+
+    // TODO: Replace UserTypes with UserRole, i.e. reconfigure authZ policy infra
+    [Authorize]
+    [HttpPatch("run/{id}")]
+    [SwaggerOperation(
+        "Updates a run with the specified new fields. This request is restricted to administrators " +
+        "or users updating their own runs. " +
+        "Note: `runType` cannot be updated. " +
+        "This operation is atomic; if an error occurs, the run will not be updated. " +
+        "All fields of the request body are optional but you must specify at least one.",
+        OperationId = "updateRun"
+    )]
+    [SwaggerResponse(204)]
+    [SwaggerResponse(401)]
+    [SwaggerResponse(403, "The user attempted to update another user's run, or the user isn't an admin.")]
+    [SwaggerResponse(404, Type = typeof(ProblemDetails))]
+    [SwaggerResponse(422, Type = typeof(ProblemDetails))]
```

Comment:
We should somehow specify that this actually can return either a problem details OR a validation problem details (in the case of an actual validation error). This issue has to be addressed for run creation as well.

Reply:
Dunno how to do this without using polymorphism, if that's even possible. I'll have to do my research.

Reply:
I guess don't worry about it then and just put a note.